arangod/Cluster/Maintenance.cpp

```diff
@@ -649,6 +663,77 @@ VPackBuilder getShardMap(VPackSlice const& collections) {
   return shardMap;
 }
 
+std::set<ShardID> extractShardsFromCollection(VPackSlice collection) {
+  auto shards = collection.get(SHARDS);
+  if (!shards.isObject()) {
+    return {};
+  }
+
+  std::set<ShardID> result;
+  for (auto shard : VPackObjectIterator(collection.get(SHARDS))) {
+    result.insert(ShardID::shardIdFromString(shard.key.stringView()).get());
+  }
+  return result;
+}
+
+std::tuple<DistributeShardsLikeMapping, ClonePrototypeMapping>
+getDistributeShardsLike(VPackSlice const& collections) {
+  DistributeShardsLikeMapping mapping;
+  ClonePrototypeMapping reverseMapping;
+
+  // maps a collection to the sorted set of shards
+  std::unordered_map<std::string_view, std::set<ShardID>> prototypeShards;
+
```

Comment:
nit: these objects are needed only after the initial `if` statement in this function.
So we can move their declarations down to after the `if`.